client/my-sites/plans-grid/components/plan-type-selector/style.scss

```diff
@@ -59,3 +60,71 @@
 		}
 	}
 }
+
+.plan-type-selector .plan-type-selector__interval-type-dropdown {
+	&,
+	&:visited,
+	&:hover span.name {
+		color: var(--color-text);
+	}
+	.components-custom-select-control__button {
+		min-width: 225px;
+	}
+	.components-custom-select-control__menu {
+		margin: 0;
+	}
+	.components-custom-select-control__item {
+		grid-template-columns: auto min-content;
+	}
+}
+
+.plans-features-main {
```

Comment:
> Given how we're approaching this, this class name shouldn't be referenced here. Nothing plans-features-main... should exist in plans-grid stylesheets, as these are consuming end concerns right now.

Ah of course. Makes total sense 👍 

> I think we will actually going to need this sticky behavior outside of Calypso though, so just moving them to plans-features-main may not be ideal. I think we may need to approach this just a tad differently. The code is fine, just the design. How about:

> 1. Migrate the wrapping with StickyContainer into the plan-type-selector component.
> 2. Pass an optional property from plans-features-main whether we want a sticky component or not e.g. isSticky (so defaults to falsy).
> 3. Rename plans-features-main__plan-type-selector-sticky-container to plan-type-selector-sticky-container

Considering this now 🤔

Reply:
> I think we will actually going to need this sticky behavior outside of Calypso though

Gotcha, so: 
- Embed plan type selector sticky container inside the `plans-grid`
- Make the plan type selector sticky behavior a setting for the `plans-grid` that consumers can enable or disable

One thing I'll note is that there's the possibility that the plan type selector dropdown will not perform well with longitudinal data analysis. I bring up this consideration because of what happens if we, as Martech, decide to revert back to the original toggle and roll back the mobile sticky behavior. The plan type selector dropdown sticky feature will be unused, but giving it more thought, with your design, we have flexibility and can simply disable sticky behavior in that case.

In reality, I doubt we'll roll back the sticky behavior ( it makes so much sense to me ), but I've been wrong plenty in the past 😆

Reply:
I started implementing this, but ran into walls 😬. Something about nesting the StickyContainer more deeply within the DOM hierarchy of the plans grid causes the container to revoke "sticky" status way too early

Reply:
Figured out the issue. @jdc91 I did some digging here and sticky positioning is always relative to its parent. When we move the `StickyContainer` from `PlansFeaturesMain` into the `IntervalTypeSelector` component, there are several wrapping components in between, with [PlanTypeSelector](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-grid/components/plan-type-selector/index.tsx#L20-L22) taking on the role of "parent".

Because of this, sticky behavior no longer ends when scrolling past the feature grid cards. Instead, it ends early when scrolling past the PlanTypeSelector container.

Mulling over how to implement this change effectively 🤔

Reply:
> I think we will actually going to need this sticky behavior outside of Calypso though, so just moving them to plans-features-main may not be ideal.

@chriskmnds Dehan has been asked to implement this feature in landpack, so this use case is now confirmed 👍